### PR TITLE
merging from main (#193)

### PR DESCRIPTION
**Pull request by Paramadon**
…192)*Issue #, if available:*
Merging main branch to nodejs-instrumention. This will allow for no/less merge conflicts when merging nodejs changes to the main branch. 
(mergeMainNodeJS) is the main branch with merge conflict changes (merge conflicts were mainly dotnet changes).
*Description of changes:*
Mainly had a bunch of Dotnet changes that needed to be pulled into this branch

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,11 @@ VERSION_PKG ?= "github.com/aws/amazon-cloudwatch-agent-operator/internal/version
 AGENT_VERSION ?= "$(shell grep -v '\#' versions.txt | grep cloudwatch-agent | awk -F= '{print $$2}')"
 AUTO_INSTRUMENTATION_JAVA_VERSION ?= "$(shell grep -v '\#' versions.txt | grep aws-otel-java-instrumentation | awk -F= '{print $$2}')"
 AUTO_INSTRUMENTATION_PYTHON_VERSION ?= "$(shell grep -v '\#' versions.txt | grep aws-otel-python-instrumentation | awk -F= '{print $$2}')"
+<<<<<<< HEAD
+AUTO_INSTRUMENTATION_DOTNET_VERSION ?= "$(shell grep -v '\#' versions.txt | grep aws-otel-dotnet-instrumentation | awk -F= '{print $$2}')"
```
Comment (mitali-salvi) [inline Makefile:9]: could you fix this merge conflict ?

File: Dockerfile [hunk 2]
```diff
@@ -26,12 +26,13 @@ ARG VERSION_DATE
...
+RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -ldflags="-X ${VERSION_PKG}.version=${VERSION} -X ${VERSION_PKG}.buildDate=${VERSION_DATE} -X ${VERSION_PKG}.agent=${AGENT_VERSION} -X ${VERSION_PKG}.autoInstrumentationJava=${AUTO_INSTRUMENTATION_JAVA_VERSION} -X ${VERSION_PKG}.autoInstrumentationPython=${AUTO_INSTRUMENTATION_PYTHON_VERSION} -X ${VERSION_PKG}.autoInstrumentationNodeJS=${AUTO_INSTRUMENTATION_NODEJS_VERSION} -X ${VERSION_PKG}.autoInstrumentationDotNet=${AUTO_INSTRUMENTATION_DOTNET_VERSION} -X ${VERSION_PKG}.dcgmExporter=${DCMG_EXPORTER_VERSION} -X ${VERSION_PKG}.neuronMonitor=${NEURON_MONITOR_VERSION}" -a -o manager main.go
```
Comment (mitali-salvi) [inline Dockerfile:35]: nitpick - could we follow the same order for application signals language support
Java
Python
DotNet
NodeJS

File: README.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
 # Amazon CloudWatch Agent Operator
 The Amazon CloudWatch Agent Operator is software developed to manage the [CloudWatch Agent](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Install-CloudWatch-Agent.html) on kubernetes.
 
+Supported Languages:
```
Comment (mitali-salvi) [inline README.md:4]: Missing support for NodeJS, lets add it as well

File: main.go [hunk 4]
```diff
@@ -48,6 +48,7 @@ const (
 	cloudwatchAgentImageRepository           = "public.ecr.aws/cloudwatch-agent/cloudwatch-agent"
 	autoInstrumentationJavaImageRepository   = "public.ecr.aws/aws-observability/adot-autoinstrumentation-java"
 	autoInstrumentationPythonImageRepository = "public.ecr.aws/aws-observability/adot-autoinstrumentation-python"
+	autoInstrumentationDotNetImageRepository = "ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-dotnet"
 	//TODO: Update with the latest release of ADOT SDK for NodeJS
```
Comment (mitali-salvi) [inline main.go:52]: This TODO can be removed

File: versions.txt [hunk 5]
```diff
@@ -7,8 +7,10 @@ operator=1.4.1
 # Represents the current release of ADOT language instrumentation.
 aws-otel-java-instrumentation=v1.32.2
 aws-otel-python-instrumentation=v0.2.0
+# This needs to be updated with the latest release of ADOT SDK
+aws-otel-dotnet-instrumentation=1.6.0
 # TODO - Update with the latest release of ADOT SDK
```
Comment (mitali-salvi) [inline versions.txt:12]: this TODO can be removed

**Review by musa-asad — APPROVED** [review]

**Review by mitali-salvi — APPROVED** [review]

**Pull request merged**